libfwupdplugin/fu-udev-device.c

```diff
@@ -1847,25 +1847,17 @@ static GBytes *
 fu_udev_device_dump_firmware(FuDevice *device, FuProgress *progress, GError **error)
 {
 	FuUdevDevice *self = FU_UDEV_DEVICE(device);
-	FuUdevDevicePrivate *priv = GET_PRIVATE(self);
 	guint number_reads = 0;
 	g_autofree gchar *fn = NULL;
 	g_autoptr(GByteArray) buf = g_byte_array_new();
 	g_autoptr(GError) error_local = NULL;
 	g_autoptr(GFile) file = NULL;
 	g_autoptr(GInputStream) stream = NULL;
-
-	/* open the file */
-	if (priv->device_file == NULL) {
-		g_set_error_literal(error,
-				    FWUPD_ERROR,
-				    FWUPD_ERROR_INTERNAL,
-				    "Unable to read firmware from device");
-		return NULL;
-	}
+	g_autofree gchar *rom_fn = NULL;
 
 	/* open file */
-	file = g_file_new_for_path(priv->device_file);
+	rom_fn = g_build_filename(fu_udev_device_get_sysfs_path(self), "rom", NULL);
```

Comment:
I'm confused; isn't `fu_optionrom_device_probe()` setting up the device file for this?

Reply:
Mm in this case I think that is overridden by amdgpu fwupd plugin?

Reply:
I've pushed a fixup which I think makes things a bit less magic; insane?

Reply:
Generally; not insane.  But I have comments below how to make it _more sane_.